include/cuco/detail/static_map.inl

```diff
@@ -22,26 +22,33 @@ template <typename Key, typename Value, cuda::thread_scope Scope, typename Alloc
 static_map<Key, Value, Scope, Allocator>::static_map(std::size_t capacity,
                                                      Key empty_key_sentinel,
                                                      Value empty_value_sentinel,
-                                                     Allocator const& alloc)
+                                                     Allocator const& alloc,
+                                                     cudaStream_t stream)
   : capacity_{std::max(capacity, std::size_t{1})},  // to avoid dereferencing a nullptr (Issue #72)
     empty_key_sentinel_{empty_key_sentinel},
     empty_value_sentinel_{empty_value_sentinel},
     slot_allocator_{alloc},
-    counter_allocator_{alloc}
+    counter_allocator_{alloc},
+    exec_stream_{stream}
 {
-  slots_         = std::allocator_traits<slot_allocator_type>::allocate(slot_allocator_, capacity_);
-  num_successes_ = std::allocator_traits<counter_allocator_type>::allocate(counter_allocator_, 1);
+  // allocator should allocate memory accessible by the exec_stream_
+  slots_         = std::allocator_traits<slot_allocator_type>::allocate(slot_allocator_, capacity_,
+                                                                        exec_stream_);
+  num_successes_ = std::allocator_traits<counter_allocator_type>::allocate(counter_allocator_, 1,
+                                                                            exec_stream_);
```

Comment:
I assumed it can given [this](https://github.com/NVIDIA/cuCollections/blob/dev/include/cuco/allocator.hpp#L35) function taking `stream` argument.

Reply:
Passing the stream to `allocator_traits` is not going to work. You need to call `allocate` directly like Yunsong showed.

Reply:
It will indeed invoke `cuco::allocator::allocate` but the stream argument won't be passed properly in this way. It doesn't matter much with `cuco::allocator` since it only takes stream but does not really use it. A real case would be using e.g. rmm allocator [allocate](https://github.com/rapidsai/rmm/blob/branch-21.12/include/rmm/mr/device/polymorphic_allocator.hpp#L82). My suggestions will work just fine.